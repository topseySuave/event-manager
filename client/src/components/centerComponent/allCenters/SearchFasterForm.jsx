import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import SelectField from 'material-ui/SelectField';
import MenuItem from 'material-ui/MenuItem';
import shortid from 'shortid';

/**
 * SearchFasterForm Class Component
 * */
class SearchFasterForm extends Component {
  /**
   * SearchFasterForm Class Coonstructor
   * @param { object } props
   * */
  constructor(props) {
    super(props);
    this.state = {
      location: '',
      price: '',
      capacity: ''
    };
    this.handleSearchChange = this.handleSearchChange.bind(this);
    this.handleSearchStart = this.handleSearchStart.bind(this);
  }

  /**
   * handleSearchChange Method
   * @param { object } e
   * @returns { void }
   * */
  handleSearchChange(e) {
    this.setState({
      [e.target.name]: e.target.value
    });
  }

  /**
   * handleSearchStart Method
   * @param { object } e
   * @returns { void }
   * */
  handleSearchStart(e) {
    e.preventDefault();
    let { location, price, capacity } = this.state;

    // if any of the search queries are provided
    // then run the search function
    if (location || price || capacity) {
      this.props.onSearch(this.state);
    }
  }

  /**
   * render Life cycle Method
   * @returns { component }
   * */
  render() {
    let {
      location, price, capacity
    } = this.state;

    return (
      <form className="col s12 full-width">
        <div className="input-field col s12 l2">
          <label htmlFor="fast_price">Price</label>
          <input
            id="fast_price"
            type="number"
            min="100"
            onChange={this.handleSearchChange}
            name="price"
            value={price || ''}
            className="validate"
          />
        </div>
        <div className="input-field col s12 l4">
          <label htmlFor="fast_capacity">Capacity</label>
          <input
            id="fast_capacity"
            type="number"
            name="capacity"
            onChange={this.handleSearchChange}
            min="0"
            value={capacity || ''}
            className="validate"
          />
        </div>
        <div className="input-field col s12 l4">
          <label htmlFor="fast_capacity">Location</label>
          <input
            id="fast_location"
            type="text"
            name="location"
            onChange={this.handleSearchChange}
            value={location || ''}
            className="validate"
          />
        </div>
        <div className="input-field col s12 l2">
          <button
            onClick={this.handleSearchStart}
            className="btn gradient__bg"
            type="submit"
          >
            search
          </button>
        </div>
      </form>
    );
  }
}

SearchFasterForm.propTypes = {
  style: PropTypes.object,
  onSearch: PropTypes.func.isRequired
};

export default SearchFasterForm;
