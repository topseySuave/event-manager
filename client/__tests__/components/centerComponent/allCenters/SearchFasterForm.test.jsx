import React from 'react';
import { shallow, mount } from 'enzyme';
import SearchFasterForm from
  '../../../../src/components/centerComponent/allCenters/searchFasterForm';

describe('SearchFasterForm component', () => {
  let props = {
    onSearch: jest.fn()
  };
  let event = {
    preventDefault: jest.fn(),
    target: {
      name: 'location',
      value: 'ikeja, lagos'
    }
  };
  const wrapper = shallow(<SearchFasterForm {...props} />);
  wrapper.setState({
    loaction: 'ikeja, lagos',
    price: 100000,
    capacity: 500
  });
  let instance = wrapper.instance();
  test('Should have a snapshot of the component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('handleSearchChange should set the state an appropiate field', () => {
    instance.handleSearchChange(event);
  });

  test('handleSearchStart should start search if all field are filled', () => {
    instance.handleSearchStart(event);
  });
});
