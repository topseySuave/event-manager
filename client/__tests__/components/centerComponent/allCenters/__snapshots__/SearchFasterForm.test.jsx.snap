// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchFasterForm component Should have a snapshot of the component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchFasterForm
    onSearch={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="input-field col s12 l2"
        >
          <label
            htmlFor="fast_price"
          >
            Price
          </label>
          <input
            className="validate"
            id="fast_price"
            min="100"
            name="price"
            onChange={[Function]}
            type="number"
            value={100000}
          />
        </div>,
        <div
          className="input-field col s12 l4"
        >
          <label
            htmlFor="fast_capacity"
          >
            Capacity
          </label>
          <input
            className="validate"
            id="fast_capacity"
            min="0"
            name="capacity"
            onChange={[Function]}
            type="number"
            value={500}
          />
        </div>,
        <div
          className="input-field col s12 l4"
        >
          <label
            htmlFor="fast_capacity"
          >
            Location
          </label>
          <input
            className="validate"
            id="fast_location"
            name="location"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>,
        <div
          className="input-field col s12 l2"
        >
          <button
            className="btn gradient__bg"
            onClick={[Function]}
            type="submit"
          >
            search
          </button>
        </div>,
      ],
      "className": "col s12 full-width",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              htmlFor="fast_price"
            >
              Price
            </label>,
            <input
              className="validate"
              id="fast_price"
              min="100"
              name="price"
              onChange={[Function]}
              type="number"
              value={100000}
            />,
          ],
          "className": "input-field col s12 l2",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Price",
              "htmlFor": "fast_price",
            },
            "ref": null,
            "rendered": "Price",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "validate",
              "id": "fast_price",
              "min": "100",
              "name": "price",
              "onChange": [Function],
              "type": "number",
              "value": 100000,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              htmlFor="fast_capacity"
            >
              Capacity
            </label>,
            <input
              className="validate"
              id="fast_capacity"
              min="0"
              name="capacity"
              onChange={[Function]}
              type="number"
              value={500}
            />,
          ],
          "className": "input-field col s12 l4",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Capacity",
              "htmlFor": "fast_capacity",
            },
            "ref": null,
            "rendered": "Capacity",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "validate",
              "id": "fast_capacity",
              "min": "0",
              "name": "capacity",
              "onChange": [Function],
              "type": "number",
              "value": 500,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              htmlFor="fast_capacity"
            >
              Location
            </label>,
            <input
              className="validate"
              id="fast_location"
              name="location"
              onChange={[Function]}
              type="text"
              value=""
            />,
          ],
          "className": "input-field col s12 l4",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Location",
              "htmlFor": "fast_capacity",
            },
            "ref": null,
            "rendered": "Location",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "validate",
              "id": "fast_location",
              "name": "location",
              "onChange": [Function],
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="btn gradient__bg"
            onClick={[Function]}
            type="submit"
          >
            search
          </button>,
          "className": "input-field col s12 l2",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "search",
            "className": "btn gradient__bg",
            "onClick": [Function],
            "type": "submit",
          },
          "ref": null,
          "rendered": "search",
          "type": "button",
        },
        "type": "div",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="input-field col s12 l2"
          >
            <label
              htmlFor="fast_price"
            >
              Price
            </label>
            <input
              className="validate"
              id="fast_price"
              min="100"
              name="price"
              onChange={[Function]}
              type="number"
              value={100000}
            />
          </div>,
          <div
            className="input-field col s12 l4"
          >
            <label
              htmlFor="fast_capacity"
            >
              Capacity
            </label>
            <input
              className="validate"
              id="fast_capacity"
              min="0"
              name="capacity"
              onChange={[Function]}
              type="number"
              value={500}
            />
          </div>,
          <div
            className="input-field col s12 l4"
          >
            <label
              htmlFor="fast_capacity"
            >
              Location
            </label>
            <input
              className="validate"
              id="fast_location"
              name="location"
              onChange={[Function]}
              type="text"
              value=""
            />
          </div>,
          <div
            className="input-field col s12 l2"
          >
            <button
              className="btn gradient__bg"
              onClick={[Function]}
              type="submit"
            >
              search
            </button>
          </div>,
        ],
        "className": "col s12 full-width",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                htmlFor="fast_price"
              >
                Price
              </label>,
              <input
                className="validate"
                id="fast_price"
                min="100"
                name="price"
                onChange={[Function]}
                type="number"
                value={100000}
              />,
            ],
            "className": "input-field col s12 l2",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Price",
                "htmlFor": "fast_price",
              },
              "ref": null,
              "rendered": "Price",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "validate",
                "id": "fast_price",
                "min": "100",
                "name": "price",
                "onChange": [Function],
                "type": "number",
                "value": 100000,
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                htmlFor="fast_capacity"
              >
                Capacity
              </label>,
              <input
                className="validate"
                id="fast_capacity"
                min="0"
                name="capacity"
                onChange={[Function]}
                type="number"
                value={500}
              />,
            ],
            "className": "input-field col s12 l4",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Capacity",
                "htmlFor": "fast_capacity",
              },
              "ref": null,
              "rendered": "Capacity",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "validate",
                "id": "fast_capacity",
                "min": "0",
                "name": "capacity",
                "onChange": [Function],
                "type": "number",
                "value": 500,
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                htmlFor="fast_capacity"
              >
                Location
              </label>,
              <input
                className="validate"
                id="fast_location"
                name="location"
                onChange={[Function]}
                type="text"
                value=""
              />,
            ],
            "className": "input-field col s12 l4",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Location",
                "htmlFor": "fast_capacity",
              },
              "ref": null,
              "rendered": "Location",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "validate",
                "id": "fast_location",
                "name": "location",
                "onChange": [Function],
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="btn gradient__bg"
              onClick={[Function]}
              type="submit"
            >
              search
            </button>,
            "className": "input-field col s12 l2",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "search",
              "className": "btn gradient__bg",
              "onClick": [Function],
              "type": "submit",
            },
            "ref": null,
            "rendered": "search",
            "type": "button",
          },
          "type": "div",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
